python/lsst/meas/base/plugins.py

```diff
@@ -475,32 +476,64 @@ def getExecutionOrder(cls):
         self.cdMatrix11Key = schema.addField(
-            "%s_CDMatrix_1_1" % name,
+            f"{name}_CDMatrix_1_1",
             type="D",
-            doc="(1, 1) element of the CDMatrix for the linear approximation "
-                "of the WCS at the src location.")
+            doc=f"(1, 1) element of the CDMatrix for the linear approximation "
+                 "of the WCS at the src location. Gives units in radians.")
         self.cdMatrix12Key = schema.addField(
-            "%s_CDMatrix_1_2" % name,
+            f"{name}_CDMatrix_1_2",
             type="D",
-            doc="(1, 2) element of the CDMatrix for the linear approximation "
-                "of the WCS at the src location.")
+            doc=f"(1, 2) element of the CDMatrix for the linear approximation "
+                 "of the WCS at the src location. Gives units in radians.")
         self.cdMatrix21Key = schema.addField(
-            "%s_CDMatrix_2_1" % name,
+            f"{name}_CDMatrix_2_1",
             type="D",
-            doc="(2, 1) element of the CDMatrix for the linear approximation "
-                "of the WCS at the src location.")
+            doc=f"(2, 1) element of the CDMatrix for the linear approximation "
+                 "of the WCS at the src location. Gives units in radians.")
         self.cdMatrix22Key = schema.addField(
-            "%s_CDMatrix_2_2" % name,
+            f"{name}_CDMatrix_2_2",
             type="D",
-            doc="(2, 2) element of the CDMatrix for the linear approximation "
-                "of the WCS at the src location.")
+            doc=f"(2, 2) element of the CDMatrix for the linear approximation "
+                 "of the WCS at the src location. Gives units in radians.")
 
     def measure(self, measRecord, exposure, center):
-        localCDMatrix = exposure.getWcs().getCdMatrix(center)
-        measRecord.set(self.cdMatrix11Key, localCDMatrix[0, 0])
-        measRecord.set(self.cdMatrix12Key, localCDMatrix[0, 1])
-        measRecord.set(self.cdMatrix21Key, localCDMatrix[1, 0])
-        measRecord.set(self.cdMatrix22Key, localCDMatrix[1, 1])
+        wcs = exposure.getWcs()
+        localMatrix = self.makeLocalTransformMatrix(wcs, center)
+        measRecord.set(self.cdMatrix11Key, localMatrix[0, 0])
+        measRecord.set(self.cdMatrix12Key, localMatrix[0, 1])
+        measRecord.set(self.cdMatrix21Key, localMatrix[1, 0])
+        measRecord.set(self.cdMatrix22Key, localMatrix[1, 1])
+
+    def makeLocalTransformMatrix(self, wcs, center):
+        """Create a local, linear approximation of the wcs transformation
+        matrix.
+
+        The approximation is created as if the center is at RA=0, DEC=0. All
+        comparing x,y coordinate are relative to the position of center. Matrix
+        is initially calculated with units arcseconds and then converted to
+        radians. This yields higher precision results due to quirks in AST.
```

Comment:
I'd move the note about using arcseconds initially to a code comment; a user of the function shouldn't care.

Reply:
It's not entirely clear to me how this can be an AST problem as such given that AST doesn't use arcseconds itself at all and does everything in radians.

Reply:
My best theory was the AST might be using the output unit of the mapping to inform the step size for finite differences, which would be reasonable (and in some contexts probably desirable) behavior even if it otherwise uses radians internally.  But it was only ever a theory, and I think the important point here is just that arcseconds work best here and we don't really care that we don't fully understand why.